registry/default/magicui/animated-subscribe-button.tsx

```diff
@@ -1,66 +1,87 @@
 "use client";
 
+import { cn } from "@/lib/utils";
+import { HTMLMotionProps } from "motion/react";
 import { AnimatePresence, motion } from "motion/react";
 import React, { useState } from "react";
 
-interface AnimatedSubscribeButtonProps {
-  buttonColor: string;
-  buttonTextColor?: string;
-  subscribeStatus: boolean;
-  initialText: React.ReactElement | string;
-  changeText: React.ReactElement | string;
+interface AnimatedSubscribeButtonProps
+  extends Omit<HTMLMotionProps<"button">, "ref"> {
+  subscribeStatus?: boolean;
+  children: React.ReactNode;
+  className?: string;
 }
 
-export const AnimatedSubscribeButton: React.FC<
+export const AnimatedSubscribeButton = React.forwardRef<
+  HTMLButtonElement,
   AnimatedSubscribeButtonProps
-> = ({
-  buttonColor,
-  subscribeStatus,
-  buttonTextColor,
-  changeText,
-  initialText,
-}) => {
-  const [isSubscribed, setIsSubscribed] = useState<boolean>(subscribeStatus);
+>(
+  (
+    { subscribeStatus = false, onClick, className, children, ...props },
+    ref,
+  ) => {
+    const [isSubscribed, setIsSubscribed] = useState<boolean>(subscribeStatus);
 
-  return (
-    <AnimatePresence mode="wait">
-      {isSubscribed ? (
-        <motion.button
-          className="relative flex h-10 w-[200px] items-center justify-center overflow-hidden rounded-md bg-white outline outline-1 outline-black"
-          onClick={() => setIsSubscribed(false)}
-          initial={{ opacity: 0 }}
-          animate={{ opacity: 1 }}
-          exit={{ opacity: 0 }}
-        >
-          <motion.span
-            key="action"
-            className="relative flex h-full w-full items-center justify-center font-semibold"
-            initial={{ y: -50 }}
-            animate={{ y: 0 }}
-            style={{ color: buttonColor }}
+    const childrenArray = React.Children.toArray(children);
+    const initialChild = childrenArray[0];
+    const changeChild = childrenArray[1];
```

Comment:
this approach is nice! we can also throw error if its not an array of 2 elements